File: kp 2020 java/src/l3/Pentagon.java
package l3;

/**
 * Creates regular pentagons
 * It extends abstract class Figure and inheritance methods surface and circumference
 *
 * @author Tobiasz Wojnar
 * execise for university course
 * @see l3.Figury for full requriments
 * @see l3.Figure
 */
public class Pentagon extends Figure {
    double side;

    /**
     * Constructor of pentagon, since we create only regular pentagons it needs only one parameter
     * @param side of pentagon, is positive, double precision number
     */
    public Pentagon (double side){//TODO how to create exception if number is not positive
        this.side=side;
    }

    /**
     * calculates surface area of regular pentagon
     * important to remember that Math.tan() functions needs angle radians
     *
     * @return surface area of regular pentagon
     */
    public double surface (){
        return 1.25*side*side /Math.tan(Math.toRadians(36));
    }

    /**
     * @return circumference of regular pentagon
     */
    public double circumference (){
        return 5*side;
    }
}

File: kp 2020 cpp/l3/Pentagon.cpp
/**
 * Creates regular pentagon
 * It extends virtual class Figure and inheritance methods surface and circumference
 *
 * @author Tobiasz Wojnar
 * excise for university course
 * @see Main.cpp for full requirements
 * @see Pentagon.hpp
 * @see Figure.hpp
 */
#include "Pentagon.hpp"
#include <cmath>

/**
 * Constructor of pentagon, since we create only regular pentagon it needs only one parameter
 * @param side of pentagon, is positive, double precision number
 */
Pentagon :: Pentagon (double side){
    this->side=side;
}
/**
 * @return surface area of regular pentagon
 */
double Pentagon :: Surface () {
    return side*side*1.25/tan(M_PI/5);
}
/**
 * @return circumference of regular pentagon
 */
double Pentagon :: Circumference () {
    return side*5;
}